### PR TITLE
fix masses and positions to match schema and report correct units

### LINKED ISSUE #285: halo mass unit is not consistent with schema (for cosmoDC2, and maybe protoDC2)
The unit halo mass in cosmoDC2 is currently Msun/h, but it should be Msun according to the schema. 

This can be easily fixed by updating the quantity modifier in the reader. @evevkovacs will help make sure that the unit is consistent in both cosmoDC2 and protoDC2 (or apply fixes if not). 

Before this is fixed, users can edit the quantity modifier directly. This way, users will not need to update their code again when the fix is applied to the reader. For example, for cosmoDC2,

```python
catalog.add_quantity_modifier(
    'halo_mass', 
    (lambda x: x / catalog.cosmology.h, 'hostHaloMass'), 
    overwrite=True
)
```

Thanks to @marina-ricci for discovering this. Thanks to @katrinheitmann and @evevkovacs for identifying the issue.

### LINKED ISSUE #287: check unit of `position` in protoDC2, baseDC2, cosmoDC2
The unit of position (x, y, z) in our schema is Mpc. The common unit used in cosmological simulations is Mpc/h. Currently the readers do not have any unit conversion, but unit of these quantities may have been converted during catalog production. 

This issue is to make sure that we double check the unit is consistent in protoDC2, baseDC2, cosmoDC2.

### PR DESCRIPTION
**Pull request by evevkovacs**
This PR should resolve the remaining units issues for cosmoDC2, baseDC2 and protoDC2 for all versions.

[edited by @yymao: this PR fixes #285 and fixes #287]

### CODE REVIEW

File: GCRCatalogs/alphaq.py [hunk 1]
```diff
@@ -357,7 +363,15 @@ def _get_native_quantity_info_dict(self, quantity, default=None):
...
             modifier = lambda k, v: None if k == 'description' and v == b'None given' else v.decode()
-            return {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            return_qty = {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            # a hot fix of the units of native halo mass (hostHaloMass) and x for v3+
+            if self.catalog_version >= StrictVersion('3.0'):
+                if quantity == 'hostHaloMass':
+                    return_qty = {'units': 'Msun/h'}
+            else:
+                if quantity == 'x' or quantity == 'y' or quantity == 'z':
+                    return_qty = {'units': 'comoving Mpc'}
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Lines 368 to 373: (i) make each hotfix indenepent (easier future maintenance) and (ii) don't overwrite the full `return_qty`, just update the unit. 
```python
            if self.catalog_version >= StrictVersion('3.0') and quantity == 'hostHaloMass':
               return_qty['units'] = 'Msun/h'
            if self.catalog_version < StrictVersion('3.0') and quantity in set('xyz'):
               return_qty['units'] = 'comoving Mpc'
```

**Comment by evevkovacs** [comment]
OK, I pushed your changes.

File: GCRCatalogs/alphaq.py [hunk 2]
```diff
@@ -357,7 +363,13 @@ def _get_native_quantity_info_dict(self, quantity, default=None):
             if quantity_key not in fh:
                 return default
             modifier = lambda k, v: None if k == 'description' and v == b'None given' else v.decode()
-            return {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            return_qty = {k: modifier(k, v) for k, v in fh[quantity_key].attrs.items()}
+            # a hot fix of the units of native halo mass (hostHaloMass) and x for v3+
+            if self.catalog_version >= StrictVersion('3.0') and quantity == 'hostHaloMass':
+                    return_qty['units'] = 'Msun/h'
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]:
```suggestion
                return_qty['units'] = 'Msun/h'
```

File: GCRCatalogs/alphaq.py [hunk 3]
```diff
@@ -357,7 +363,13 @@ def _get_native_quantity_info_dict(self, quantity, default=None):
...
+            if self.catalog_version < StrictVersion('3.0') and quantity in set('xyz'):
+                    return_qty['units'] = 'comoving Mpc'
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]:
```suggestion
                return_qty['units'] = 'comoving Mpc'
```

**Comment by yymao** [comment]
@evevkovacs I took the liberty of fixing the final minor issue. This is now ready to merge at your convenience.

**Pull request merged**